src/autobattle.h

```diff
@@ -0,0 +1,193 @@
+public:
+	static constexpr auto name = "RPG_RT+";
+
+	StringView GetName() const override { return name; }
+private:
+	void vSetAutoBattleAction(Game_Actor& source) override;
+};
+
+/**
+ * Calcuate the auto battle effectiveness rank of source using healing skill on target.
+ *
+ * @param source the user of the skill
+ * @param target the target of the skill
+ * @param skill the skill
+ * @param apply_variance If true, apply variance to the damage
+ * @param emulate_bugs Emulate all RPG_RT bugs for accuracy
+ *
+ * @pre skill Must be a normal or subskill or the result is undefined.
+ * @pre skill must target self, ally, or ally party or the result is undefined.
+ */
+double CalcSkillHealAutoBattleTargetRank(const Game_Actor& source, const Game_Battler& target, const lcf::rpg::Skill& skill, bool apply_variance, bool emulate_bugs);
+
+/**
+ * Calcuate the auto battle effectiveness rank of source using damage skill on target.
+ *
+ * @param source the user of the skill
+ * @param target the target of the skill
+ * @param skill the skill
+ * @param apply_variance If true, apply variance to the damage
+ * @param emulate_bugs Emulate all RPG_RT bugs for accuracy
+ *
+ * @pre skill Must be a normal or subskill or the result is undefined.
+ * @pre skill must target enemy, or enemy party or the result is undefined.
+ */
+double CalcSkillDmgAutoBattleTargetRank(const Game_Actor& source, const Game_Battler& target, const lcf::rpg::Skill& skill, bool apply_variance, bool emulate_bugs);
+
+/**
+ * Calcuate the auto battle total effectiveness rank of using a skill.
+ *
+ * @param source the user of the skill
+ * @param skill the skill
+ * @param apply_variance If true, apply variance to the damage
+ * @param emulate_bugs Emulate all RPG_RT bugs for accuracy
+ */
+double CalcSkillAutoBattleRank(const Game_Actor& source, const lcf::rpg::Skill& skill, bool apply_variance, bool emulate_bugs);
+
+/**
+ * Calcuate the auto battle effectiveness rank of source attacking target.
+ *
+ * @param source the user of the skill
+ * @param target the target of the skill
+ * @param cond the battle condition
+ * @param apply_variance If true, apply variance to the damage
+ * @param emulate_bugs Emulate all RPG_RT bugs for accuracy
+ */
+double CalcNormalAttackAutoBattleTargetRank(const Game_Actor& source, const Game_Battler& target, Game_Battler::Weapon weapon, lcf::rpg::System::BattleCondition cond, bool apply_variance, bool emulate_bugs);
+
+/**
+ * Calcuate the auto battle total effectiveness rank of using a normal attack.
+ *
+ * @param source the user of the skill
+ * @param weapon Which weapon to use in the calculation
+ * @param cond the battle condition
+ * @param apply_variance If true, apply variance to the damage
+ * @param emulate_bugs Emulate all RPG_RT bugs for accuracy
+ */
+double CalcNormalAttackAutoBattleRank(const Game_Actor& source, Game_Battler::Weapon weapon, const lcf::rpg::System::BattleCondition cond, bool apply_variance, bool emulate_bugs);
```

Comment:
"Clang-Tidy: Parameter 'cond' is const-qualified in the function declaration; const-qualification of parameters only has an effect in function definitions" whatever that means ;)

Reply:
You should never add const to pass by value parameter in a declaration because it does nothing.

Reply:
oh I didn't see that this is "by value". Its getting late. Right, makes sense.